app/controllers/questions_controller.rb

```diff
@@ -0,0 +1,57 @@
+class QuestionsController < ApplicationController
+
+  def index
+    @questions = Question.all
+  end
+
+  def new
+    if logged_in?
+      @question = Question.new
+    else
+      redirect_to root_path
+    end
+  end
+
+  def create
+    @question = current_user.questions.new(question_params)
+
+    if @question.save
+      @tags = tag_params.to_a.flatten
+      @tags[1].gsub( ',', '').split(' ').each do |tag|
+        @tag = Tag.find_by(title: tag)
+        if @tag
+          @tag.question_tags.create(question: @question)
+        else
+          @new_tag = Tag.create(title: tag).question_tags.create(question: @question)
+        end
+      end
```

Comment:
This is some pretty complicated code to be living in the controller. We want out controllers to be simple and clean. As soon as code gets hairy you should be asking "is there a better place for this?". Fat model, skinny controller is a bit of a mantra.